worlds/shivers/Rules.py

```diff
@@ -0,0 +1,224 @@
+            "Ixupi Captured Cloth": lambda state: cloth_capturable(state, player),
+            "Ixupi Captured Wood": lambda state: wood_capturable(state, player),
+            "Ixupi Captured Crystal": lambda state: crystal_capturable(state, player),
+            "Ixupi Captured Sand": lambda state: sand_capturable(state, player),
+            "Ixupi Captured Metal": lambda state: metal_capturable(state, player),
+            "Final Riddle: Planets Aligned": lambda state: state.can_reach("Fortune Teller", "Region", player),
+            "Final Riddle: Norse God Stone Message": lambda state: (state.can_reach("Fortune Teller", "Region", player) and state.can_reach("UFO", "Region", player)),
+            "Final Riddle: Beth's Body Page 17": lambda state: beths_body_available(state, player),
+            "Final Riddle: Guillotine Dropped": lambda state: beths_body_available(state, player),
+            },
+        "locations_puzzle_hints": {
+            "Puzzle Solved Clock Tower Door": lambda state: state.can_reach("Three Floor Elevator", "Region", player),
+            "Puzzle Solved Clock Chains": lambda state: state.can_reach("Bedroom", "Region", player),
+            "Puzzle Solved Tiki Drums": lambda state: state.can_reach("Clock Tower", "Region", player),
+            "Puzzle Solved Red Door": lambda state: state.can_reach("Maintenance Tunnels", "Region", player),
+            "Puzzle Solved UFO Symbols": lambda state: state.can_reach("Library", "Region", player),
+            "Puzzle Solved Maze Door": lambda state: state.can_reach("Projector Room", "Region", player),
+            "Puzzle Solved Theater Door": lambda state: state.can_reach("Underground Lake", "Region", player),
+            "Puzzle Solved Columns of RA": lambda state: state.can_reach("Underground Lake", "Region", player),
+            "Final Riddle: Guillotine Dropped": lambda state: state.can_reach("Underground Lake", "Region", player)
+            },
+        "elevators": {
+            "Puzzle Solved Underground Elevator": lambda state: ((state.can_reach("Underground Lake", "Region", player) or state.can_reach("Office", "Region", player)
+                                                                  and state.has("Key for Office Elevator", player))),
+            "Puzzle Solved Bedroom Elevator": lambda state: (state.can_reach("Office", "Region", player) and state.has_all({"Key for Bedroom Elevator","Crawling"}, player)),
+            "Puzzle Solved Three Floor Elevator": lambda state: ((state.can_reach("Maintenance Tunnels", "Region", player) or state.can_reach("Blue Maze", "Region", player)
+                                                                  and state.has("Key for Three Floor Elevator", player)))
+            },
+        "lightning": {
+            "Ixupi Captured Lightning": lambda state: lightning_capturable(state, player)
+        }
+    }
+    return rules_lookup
+
+
+def set_rules(Shivers: World) -> None:
```

Comment:
Minor nitpick, but should probably be either `shivers: World` or `world: World` to match PEP8.

Although that being said, if you want to have good type hinting, add to your imports:

```
from typing import TYPE_HINTING  # partial for example

if TYPE_HINTING:
	from . import ShiversWorld
```

Then use `def set_rules(world: "ShiversWorld") -> None:` (quotes are needed since you aren't actually importing `ShiversWorld`.)

Then you get nice type hinting when you access the `options` attribute. I would recommend this over what I said above.

Reply:
Updated to get good type hinting as suggested.